RandLAPACK/drivers/rl_revd2.hh

```diff
@@ -113,61 +113,59 @@ class REVD2 : public REVD2alg<T> {
 };
 
 // -----------------------------------------------------------------------------
-
+/// Power scheme for error estimation, based on algorithm E.1 from https://arxiv.org/pdf/2110.02820.pdf.
+/// This routine is too specialized to be included into RandLAPACK::utils
+/// p - number of algorithm iterations
+/// vector_buf - buffer for vector operations
+/// Mat_buf - buffer for matrix operations
+/// All other parameters come from REVD2
 template <typename T>
 T power_error_est(
     int64_t m,
     int64_t k,
     int p,
-    T* Omega_dat,
+    T* vector_buf_dat,
     T* V_dat,
     blas::Uplo uplo,
     T* A_dat,
-    T* Y_dat,
+    T* Mat_buf_dat,
     T* eigvals_dat
 ) {
     T err = 0;
-
-            char name [] = "A";
-        RandBLAS::util::print_colmaj(m, m, A_dat, name);
         
     for(int i = 0; i < p; ++i) {
-        T g_norm = blas::nrm2(m, Omega_dat, 1);
+        T g_norm = blas::nrm2(m, vector_buf_dat, 1);
         // Compute g = g / ||g|| - we need this because dot product does not take in an alpha
-        std::for_each(Omega_dat, Omega_dat + m, [g_norm](T &entry) { entry /= g_norm;});
+        std::for_each(vector_buf_dat, vector_buf_dat + m, [g_norm](T &entry) { entry /= g_norm;});
```

Comment:
I think I might be missing something. Can't this be done with ``blas::scal``?

Reply:
Yes, it can be. I'm just no using scal function very often, so didnt think about that.